protos/orchestrator_service.proto

```diff
@@ -380,6 +380,7 @@ message PurgeInstancesRequest {
         string instanceId = 1;
         PurgeInstanceFilter purgeInstanceFilter = 2;
     }
+    bool recursive = 3;
```

Comment:
If the [default](https://github.com/microsoft/durabletask-go/pull/47#discussion_r1424792139) is to enable recursive purging/termination, consider using a boolean such as "no_recursive"? Ideally default values are the zero values for the type

Reply:
I agree that this is the ideal way to go, but I'm also a bit worried about the change of behavior for existing client SDKs (for example, even Durable Functions uses this protobuf for its Java and .NET Isolated SDKs). Happy to revisit this in a separate PR.